tests/system/python/test_e2e_kafka.py

```diff
@@ -0,0 +1,207 @@
+
+        # Call the start south service fixture
+        start_south(SOUTH_PLUGIN_NAME, south_branch, foglamp_url, service_name=SOUTH_PLUGIN_NAME)
+
+        # Call the configure and start north task
+        self._configure_and_start_north_kafka(north_branch, foglamp_url, kafka_host, kafka_port, kafka_topic)
+
+        # Provide the fixture value
+        yield self.start_south_north
+
+        # Cleanup code that runs after the caller test is over
+        remove_data_file(fogbench_template_path)
+        remove_directories("/tmp/foglamp-south-{}".format(SOUTH_PLUGIN_NAME))
+        remove_directories("/tmp/foglamp-north-{}".format(NORTH_PLUGIN_NAME.lower()))
+
+    def test_end_to_end(self, start_south_north, foglamp_url, wait_time, kafka_host, kafka_rest_port, kafka_topic):
+        """ Test that data is inserted in FogLAMP and sent to Kafka
+            start_south_north: Fixture that starts FogLAMP with south and north instance
+            Assertions:
+                on endpoint GET /foglamp/asset
+                on endpoint GET /foglamp/asset/<asset_name>
+                data received from Kafka is same as data sent"""
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        time.sleep(wait_time)
+        subprocess.run(["cd $FOGLAMP_ROOT/extras/python; python3 -m fogbench -t ../../data/{}; cd -"
+                       .format(FOGBENCH_TEMPLATE)], shell=True, check=True)
+        time.sleep(wait_time)
+        conn.request("GET", '/foglamp/asset')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        assert 1 == len(val)
+        assert ASSET_NAME == val[0]["assetCode"]
+        assert 1 == val[0]["count"]
+
+        conn.request("GET", '/foglamp/asset/{}'.format(ASSET_NAME))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        val = json.loads(r)
+        assert 1 == len(val)
+        assert {'sensor': SENSOR_VALUE} == val[0]["reading"]
+
+        self._read_from_kafka(kafka_host, kafka_rest_port, kafka_topic)
+
+    def _read_from_kafka(self, host, rest_port, topic):
+        conn = http.client.HTTPConnection("{}:{}".format(host, rest_port))
+
+        # Assertions on Kafka topic, consumer and subscription
+        # GET topics
+        self._verify_kafka_topic(conn, topic)
+
+        # GET topic by name
+        self._verify_kafka_topic_by_name(conn, topic)
+
+        # Create a consumer for JSON data, starting at the beginning of the topic's
+        # The consumer group is called "CONSUMER_GROUP" and the instance is "CONSUMER_INSTANCE".
+        self._verify_kafka_consumer_group_and_instance(conn)
+
+        # Subscribe the consumer to a topic
+        self._verify_consumer_subscription_to_topic(conn, topic)
+
+        # Then consume some data from a topic using the base URL in the first response.
+        self._verify_consume_data_from_topic(conn)
```

Comment:
recreate the connection inside _verify_consume_data_from_topic, pass only the host and rest_port!

and make sure to pass header as in DELETE and POST else you will see de-serialization error

Reply:
browser request: 

http://127.0.0.1:8082/consumers/foglamp_consumer/instances/foglamp_instance/records

response: 
`{"error_code":40601,"message":"The requested embedded data format does not match the deserializer for this consumer instance"}`